pulser-core/pulser/sequence/sequence.py

```diff
@@ -1149,8 +1149,14 @@ def enable_eom_mode(
             detuning_on = cast(float, detuning_on)
             eom_config = cast(RydbergEOM, channel_obj.eom_config)
             if not isinstance(optimal_detuning_off, Parametrized):
-                detuning_off = eom_config.calculate_detuning_off(
-                    amp_on, detuning_on, optimal_detuning_off
+                (
+                    detuning_off,
+                    switching_beams,
+                ) = eom_config.calculate_detuning_off(
+                    amp_on,
+                    detuning_on,
+                    optimal_detuning_off,
+                    return_switching_beams=True,
```

Comment:
I really find it that it would make sense to have switching beams a property, it would be also way closer to the fact that you are defining it as an attribute in EOMSettings

Reply:
I really don't get what class you are referring to... The switching beams returned by `calculate_detuning_off()` actually depend on the inputs (when there is more than one controlled beam). They say: "for this combination of `amp_on` and `detuning_on`, what beam(s) should I be switching so that my `detuning_off` is as close as possible to `optimal_detuning_off`?"

Reply:
~Yes, I get this, but you have two switching_beams: the one from calculate_detuning_off and the one from detuning_off_options. I propose to place what you use in detuning_off_options in a separate property that you can use here by doing switching_beams[det_off]~

Okay now I see why you want to return switching_beams in calculate_detuning_off and I agree it's good to have it there (I thought the user had access to closest_option in which case eom_config.switching_beams[closest_option] would be easy). I am just afraid we are doing something overly complex/hard to maintain or build above with these functions having two return types, that's why I am still suggesting having a property for switching_beams in detuning_off_options ?

Reply:
I just pushed the changes. Not a big fan of having the switching beams combinations defined independently but I guess it's okay, it did allow to do some nice simplifications